### PR TITLE
Add support for .graphqlconfig

### PR DESCRIPTION
**Pull request by lydell**
This adds .graphqlconfig support, fixing #235. Notes:

- The implementation is basically copy-paste from https://github.com/apollographql/eslint-plugin-graphql/blob/45b3f95dcdaa5154233c10fb90a21da20c8ad1f5/src/index.js#L265-L274

- This should be backwards compatible. Previously, the `--schema` option was required and defaulted to `schema.json`. Now it is not "required" anymore and works like this:

  1. If specified, the `--schema` option wins.
  2. Otherwise we look for a `.graphqlconfig` file.
  3. Finally we look for a `schema.json` file.
  3. If nothing can be found, an error is thrown explaining that a schema is required and how to provide one.

  So it should be backwards compatible.

- I have not added tests yet, because I wanted to ask if you think this is the right approach first. Also, **I'm unsure where you'd like me to add the tests** and **how:** unit test the `loadSchemaFromConfig` function?

- I also get this error when building the TypeScript files. The files are still compiled somehow, because I've been able to test this PR manually by creating a `.graphqlconfig` file an running `node ./lib/cli.js generate`. I'm new to TypeScript, so I'm unsure how to fix this.

```
$ npm run compile

> apollo-codegen@0.17.0-alpha.1 compile /home/lydell/forks/apollo-codegen
> tsc

node_modules/graphql-request/dist/src/types.d.ts(5,14): error TS2304: Cannot find name 'RequestInit'.
node_modules/graphql-request/dist/src/types.d.ts(9,12): error TS2304: Cannot find name 'RequestInit'.
node_modules/graphql-request/dist/src/types.d.ts(10,19): error TS2304: Cannot find name 'RequestInit'.
node_modules/graphql-request/dist/src/types.d.ts(11,13): error TS2304: Cannot find name 'RequestInit'.
node_modules/graphql-request/dist/src/types.d.ts(12,16): error TS2304: Cannot find name 'RequestInit'.
node_modules/graphql-request/dist/src/types.d.ts(13,16): error TS2304: Cannot find name 'RequestInit'.
node_modules/graphql-request/dist/src/types.d.ts(14,22): error TS2304: Cannot find name 'RequestInit'.
node_modules/graphql-request/dist/src/types.d.ts(15,17): error TS2304: Cannot find name 'RequestInit'.
```

### CODE REVIEW

**Comment by lydell** [comment]
**TL;DR** It works! (I just tested using this PR in a real project.) Just need some advice to get across the finish line.

**Comment by martijnwalraven** [comment]
Thanks a lot for working on this! 

I'm ok with not having any new tests for this, because we're not really testing the CLI anyway (we probably should...). Also, we'll probably want to extract the CLI code into a common package so we don't have to copy/paste code between the different projects.

The TypeScript error turned out to be pretty hairy, and the only easy workaround was to add the `dom` lib to `tsconfig.json`. Hopefully we can solve this soon upstream (see https://github.com/graphcool/graphql-request/issues/15).

**Comment by lydell** [comment]
Awesome! Let me know if there's anything you'd like me to do before merging.

**Comment by lydell** [comment]
Merge conflicts fixed. Let me know if there's anything else  you'd like me to do before merging.

**Comment by lydell** [comment]
Merge conflicts fixed (again!). Let me know if there's anything else you'd like me to do before merging.

**Comment by martijnwalraven** [comment]
Sorry, I seem to have missed this was ready to merge before. Thanks for working on this!
